### PR TITLE
perf: always check data when looking for last char of needle

### PR DESCRIPTION
**Pull request by gurgunday**
Forgot this one

Since we've now established that:

- The lookbehind length is at max `needleLength - 1`, we have in the beginning `pos = -lookbehind_size >= -(needleLength - 1)`
- And since we start by checking the last character, that last char will always be in `data` as `pos + needleLength - 1 >= 0`, and pos values equal to and larger than 0 correspond to the incoming data rather than the lookbehind, so we can safely remove the method

In short, `ch` will always be in `data`

### CODE REVIEW

**Review by zekth — APPROVED** [review]
:v:
